operators/dream_texture.py

```diff
@@ -233,25 +230,28 @@ def image_callback(shared_memory_name, seed, width, height, upscaled=False):
             exception_callback=handle_exception
         )
         context.window_manager.modal_handler_add(self)
-        self.timer = context.window_manager.event_timer_add(1 / 15, window=context.window)
+        global timer
+        timer = context.window_manager.event_timer_add(1 / 15, window=context.window)
         return {'RUNNING_MODAL'}
 
-def remove_timer(context):
+def modal_stopped(context):
     global timer
     if timer:
         context.window_manager.event_timer_remove(timer)
         timer = None
-
-def kill_generator(context=bpy.context):
-    GeneratorProcess.kill_shared()
-    remove_timer(context)
-    bpy.context.scene.dream_textures_progress = 0
-    bpy.context.scene.dream_textures_info = ""
+    if not hasattr(context,'scene'):
+        context = bpy.context # modal context is sometimes missing scene?
```

Comment:
That's odd, since the scene is supposed to be in the global context, so all other contexts would be built on that (i assumed)

Reply:
I may have mixed up where it was getting context from, or completely forgetting what caused the issue in the first place. There actually seems to be a problem with the context that `kill_generator()` gives it, the default context argument is somehow different from `bpy.context` when called. It should stop being an issue if I just leave the default as `None` and replace with `bpy.context` when that is the case.